src/v/pandaproxy/json/requests/fetch.h

```diff
@@ -42,20 +42,26 @@ class rjson_serialize_impl<model::record> {
       , _tpv(tpv)
       , _base_offset(base_offset) {}
 
-    void
+    bool
     operator()(::json::Writer<::json::StringBuffer>& w, model::record record) {
         w.StartObject();
         w.Key("topic");
         ::json::rjson_serialize(w, _tpv.topic);
         w.Key("key");
-        rjson_serialize_fmt(_fmt)(w, record.release_key());
+        if (!rjson_serialize_fmt(_fmt)(w, record.release_key())) {
+            throw serialize_error("Unable to serialize key value of record");
+        }
         w.Key("value");
-        rjson_serialize_fmt(_fmt)(w, record.release_value());
+        if (!rjson_serialize_fmt(_fmt)(w, record.release_value())) {
+            throw serialize_error("Unable to deserialize key value of record");
```

Comment:
This error message doesn't match the one for key, I don't know which if any is wrong. Just do as the other proxy, I guess.

Reply:
Yeah the error messages were badly written, my bad.

As for compatibility, the error coming back is more verbose than what we can provide, but I'll see if I can get more information into the message.